Added Machine ID guide for database access from custom apps (#13226)
gravitational/teleport: docs/pages/machine-id/reference/cli.mdx

# This shows `--help` for `tsh db`:
$ tbot db --proxy=example.com --destination-dir=./tbot-user/ -- --help
```

## `tbot proxy`

If you want to access Teleport resources on a cluster using
[TLS Routing](../../setup/operations/tls-routing.mdx), you'll need to 
run access the resources through a local proxy. The `tbot proxy` command wraps
`tsh proxy` to provide local proxy functionality for various protocols,
including SSH and database access.

Note that `tsh` must also be installed to make use of this command.

| Flag                | Description                                                                                              |
|---------------------|----------------------------------------------------------------------------------------------------------|
| `-d/--debug`        | Enable verbose logging to stderr.                                                                        |
| `-c/--config`       | Path to a configuration file.                                                                            |
| `--destination-dir` | Path to the Machine ID destination dir that should be used for authentication. Required.                 |

> | `--proxy`           | The `host:port` of the Teleport proxy to use to access resources. Required.                              |

Author: Yes, I'll rephrase

# This shows `--help` for `tsh proxy`:
$ tbot proxy --proxy=example.com --destination-dir=./tbot-user/ -- --help
```

### `tbot proxy ssh` for SSH access

This forwards standard input and output over a proxy suitable for use as an
OpenSSH `ProxyCommand`.

```code
$ tbot proxy --destination-dir=./tbot-user --proxy=proxy.example.com:3080 ssh alice@node:3022
```

In this case:
 * `alice` is the remote username
 * `node` is the Teleport node name
 * `3022` is the remote SSH port, which is 3022 for nodes running the Teleport
   SSH service.

You can also refer to Machine ID's generated `ssh_config` for a proxy usage
example.

### `tbot proxy db` for database access

This opens a local proxy server to the given database. Your database client
must still be configured with client TLS certificates

```code
$ tbot proxy --destination-dir=./tbot-user --proxy=proxy.example.com:3080 db --port=1234 example
```

In this case:
 * `example` is the name of the database server as it exists in Teleport
 * `1234` is an arbitrary port on which to run the proxy

Though not recommended, to avoid the need to additional client authentication,

> the `--tunnel` flag may be used to perform authentication at the proxy:

Author: I'll rephrase this.

    --bot-user=teleport \
    --reader-user=jenkins
```

## `tbot db`

Retrieves basic information about and connects to databases using native
clients. 

This is best used for testing and validation purposes; most

> use-cases will likely prefer to connect their own databases to a local proxy

Reviewer: ```suggestion
users will likely prefer to connect their own databases to a local proxy
```

using `tbot proxy db` as described below.

Note that `tsh` must also be installed to make use of this command.

| Flag                | Description                                                                                              |
|---------------------|----------------------------------------------------------------------------------------------------------|
| `-d/--debug`        | Enable verbose logging to stderr.                                                                        |
| `-c/--config`       | Path to a configuration file.                                                                            |
| `--destination-dir` | Path to the Machine ID destination dir that should be used for authentication. Required.                 |

> | `--proxy`           | The `host:port` of the Teleport proxy to use to access resources. Required.                              |

Reviewer: ```suggestion
| `--proxy`           | The `host:port` of the Teleport Proxy Service to use to access resources. Required.                              |
```
To distinguish this from the local proxy

OpenSSH `ProxyCommand`.

```code
$ tbot proxy --destination-dir=./tbot-user --proxy=proxy.example.com:3080 ssh alice@node:3022
```

In this case:
 - `alice` is the remote username
 - `node` is the Teleport Node name
 - `3022` is the remote SSH port, which is 3022 for Nodes running the Teleport
   SSH service.

You can also refer to Machine ID's generated `ssh_config` for a proxy usage
example.

### `tbot proxy db` for database access

This opens a local proxy server to the given database. Your database client
must still be configured with client TLS certificates

```code
$ tbot proxy --destination-dir=./tbot-user --proxy=proxy.example.com:3080 db --port=1234 example
```

In this case:
 - `example` is the name of the database server as it exists in Teleport
 - `1234` is an arbitrary port on which to run the proxy

Though not recommended, to avoid the need for additional client authentication,
the `--tunnel` flag may be used to perform authentication at the local proxy
rather than within your client:

```code
$ tbot proxy --destination-dir=./tbot-user --proxy=proxy.example.com:3080 db --tunnel --port=1234 example
```

Note that this decreases security:

>  - It allows any user on the system to access the database via `localhost`

Reviewer: ```suggestion
 - It allows any user on the system to access the database via `localhost`.
```

 - Your connection to the database will be unencrypted until it reaches the

>    tbot proxy running on `localhost`.

Reviewer: ```suggestion
   `tbot` proxy running on `localhost`.
```
